xarray/backends/common.py

```diff
@@ -143,7 +146,11 @@ def add(self, source, target):
             self.sources.append(source)
             self.targets.append(target)
         else:
-            target[...] = source
+            try:
+                target[...] = source
+            except TypeError:
+                # workaround for GH: scipy/scipy#6880
+                target[slice(None, None, None)] = source
```

Comment:
Yes, this seems reasonable to me. I would just write `target[:] = source`, though -- no need to manually make the `slice()` object. This should work because variables with unlimited dimensions will never be scalars.